subsys/bluetooth/host/smp.c

```diff
@@ -2601,10 +2611,18 @@ static u8_t smp_pairing_rsp(struct bt_smp *smp, struct net_buf *buf)
 
 	smp->method = get_pair_method(smp, rsp->io_capability);
 
-	if (IS_ENABLED(CONFIG_BT_SMP_SC_ONLY) && smp->method == JUST_WORKS) {
+	if ((IS_ENABLED(CONFIG_BT_SMP_SC_ONLY) ||
+	     conn->required_sec_level == BT_SECURITY_FIPS) &&
+	     smp->method == JUST_WORKS) {
 		return BT_SMP_ERR_AUTH_REQUIREMENTS;
 	}
 
+	if ((IS_ENABLED(CONFIG_BT_SMP_SC_ONLY) ||
+	     conn->required_sec_level == BT_SECURITY_FIPS) &&
+	     get_encryption_key_size(smp) != BT_SMP_MAX_ENC_KEY_SIZE) {
+		return BT_SMP_ERR_ENC_KEY_SIZE;
+	}
+
```

Comment:
The checks here and in smp_pairing_req are quite similar, I wonder if wouldn't make sense to make get_pair_method to do these checks directly. Also Id assume required_sec_level should always be BT_SECURITY_FIPS if SC_ONLY is enabled, so perhaps required_sec_level is not being initialized properly.

Reply:
"Also Id assume required_sec_level should always be BT_SECURITY_FIPS if SC_ONLY is enabled, so perhaps required_sec_level is not being initialized properly."
required_sec_level is initialized properly.
 I added  || req == FIPS  for the case when SC_ONLY is not enabled, but bt_security(FIPS) was called. I want it to behave in the same way then.

Your other point, the code in req/rsp is quite similar, but there are some differences. How much would you suggest we make common?